static/js/containers/pages/b2b/B2BPurchasePage.js

```diff
@@ -120,26 +122,39 @@ export class B2BPurchasePage extends React.Component<Props, State> {
 
     return (
       <React.Fragment>
-        <B2BPurchaseForm
-          onSubmit={this.onSubmit}
-          products={products.filter(
-            product => product.visible_in_bulk_form === true
-          )}
-          checkout={checkout}
-          couponStatus={couponStatus}
-          contractNumber={contractNumber}
-          clearCouponStatus={clearCouponStatus}
-          fetchCouponStatus={fetchCouponStatus}
-          requestPending={requestPending}
-          productId={productId}
-          discountCode={discountCode}
-        />
+        {isLoading ? (
+          <div className="page page-loader text-center align-self-center">
+            <div className="loader-area">
+              <img
+                src="/static/images/loader.gif"
+                className="mx-auto d-block"
+              />
+              One moment while we prepare bulk purchase page
+            </div>
+          </div>
+        ) : (
+          <B2BPurchaseForm
+            onSubmit={this.onSubmit}
+            products={products.filter(
+              product => product.visible_in_bulk_form === true
+            )}
+            checkout={checkout}
+            couponStatus={couponStatus}
+            contractNumber={contractNumber}
+            clearCouponStatus={clearCouponStatus}
+            fetchCouponStatus={fetchCouponStatus}
+            requestPending={requestPending}
+            productId={productId}
+            discountCode={discountCode}
+          />
+        )}
       </React.Fragment>
     )
   }
 }
 
 const mapStateToProps = state => ({
+  isLoading:      state.entities.products ? false : true,
```

Comment:
Could you switch this to `isPending` property of the query instead since that's more reflective of the query state?